### ISSUE TITLE
[Research] Brain Health Score from Sleep EEG via Philosopher's Stone Model

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#131]
## Research Topic
Using deep learning on overnight sleep EEG to compute a single Brain Health Score that predicts cognition, disease risk, and mortality — leveraging the open-source Philosopher's Stone model from Harvard/MIT.

## Paper References

1. **"Brain Health from Sleep EEG: A Multicohort, Deep Learning Biomarker for Cognition, Disease, and Mortality"** — Sun et al., NEJM AI, 2025. Multi-task deep neural network trained on **36,000 polysomnography recordings from 27,000 subjects across 6 cohorts** (FHS, MESA, MrOS, SOF, KoGES, MGH). Produces a 1024-dimensional brain health latent space and a single Brain Health Score from **a single EEG channel**. Key results: one-standard-deviation improvement in Brain Health Score associated with **31-35% reduced mortality risk**. Outperforms demographic and expert-defined EEG feature models on all endpoints.

2. **"Machine learning of brain-specific biomarkers from EEG"** — Engemann et al., eBioMedicine (Lancet), August 2024. Framework for distinguishing CNS-specific vs. peripheral signals in EEG-based ML. Analyzed 2,600+ recordings from TUAB and TDBRAIN databases. Key finding: basic artifact rejection improves brain age prediction, but aggressive ICA-based artifact removal DECREASES performance — peripheral signals (EMG, EOG) carry useful diagnostic information. Brain age prediction achievable from minimal channels.

3. **"Evaluating Cognitive Decline Detection in Aging Populations with Single-Channel EEG"** — Scientific Reports, 2025. Meta-analysis demonstrating cognitive decline detection from **single-channel EEG** power spectral density features. Directly validates that minimal-channel recording (like Muse 2) can capture clinically meaningful brain health signals.

4. **"Dementia Risk and Machine Learning-Derived Brain Age Index from Sleep EEG"** — PMC, 2025. Pooled analysis of 7,000+ individuals across 5 community cohorts. Sleep EEG-derived brain age index predicts dementia risk years before clinical onset. Confirms sleep EEG as a window into brain health.

## Application to NeuralDreamWorkshop

The system already has sleep staging (92.98% accuracy) and records overnight EEG sessions. The Philosopher's Stone model is **open-source on GitHub** (bdsp-core/philosophers-stone) with a Python API, making integration straightforward.

### Integration path:
1. Record overnight sleep EEG via Muse 2 (already supported — sleep staging page exists)
2. Extract single-channel data (AF7 or AF8 — best signal quality frontal channels)
3. Pass through Philosopher's Stone model → get 1024-D embedding + Brain Health Score
4. Display on Brain Health dashboard with longitudinal tracking

### New files:
- `ml/models/brain_health_scorer.py` — BrainHealthScorer wrapper class
- `ml/training/prepare_sleep_eeg.py` — format Muse recordings for PS model input
- `client/src/pages/brain-health-score.tsx` — Brain Health Score dashboard

### Key classes/methods:
```python
class BrainHealthScorer:
    """Wraps the Philosopher's Stone model for Brain Health scoring."""
    
    def __init__(self):
        self.ps_model = None  # loaded from philosophers-stone package
        self.min_recording_hours = 4  # minimum for reliable score
    
    def prepare_recording(self, eeg_overnight: np.ndarray, fs: int = 256) -> np.ndarray:
        """Prepare Muse 2 overnight recording for PS model.
        - Select best frontal channel (AF7 or AF8 based on signal quality)
        - Resample if needed (PS expects specific sampling rate)
        - Apply minimal preprocessing (bandpass only)
        Returns: single-channel prepared EEG"""
    
    def compute_brain_health(self, prepared_eeg: np.ndarray) -> dict:
        """Returns: brain_health_score (0-100), latent_embedding (1024-D),
                    predicted_cognitive_score, disease_risk_profile,
                    confidence, recording_quality_grade"""
    
    def track_longitudinal(self, scores: list[dict]) -> dict:
        """Track Brain Health Score over weeks/months.
        Returns: trend (improving/stable/declining), rate_of_change,
                 percentile_for_age, recommendation"""
```

### API endpoints:
- `POST /brain-health/analyze-sleep` — process overnight recording
- `GET /brain-health/score` — latest Brain Health Score
- `GET /brain-health/trend` — longitudinal trend analysis
- `GET /brain-health/report` — comprehensive brain health report

### Frontend:
- Brain Health Score gauge (0-100) with age-matched percentile
- 1024-D embedding visualization (t-SNE/UMAP projection)
- Longitudinal trend chart (score over weeks/months)
- Sleep quality correlation with Brain Health Score

## Expected Accuracy/Impact
- Brain Health Score: validated across 36,000 recordings, 6 cohorts — the most robust EEG-based brain health metric published to date
- Mortality prediction: **31-35% risk reduction per 1-SD score improvement** (NEJM AI)
- Single-channel requirement: perfect for Muse 2 — only needs AF7 or AF8
- Caveat: model trained on clinical PSG data (silver chloride electrodes) — Muse 2 dry electrode signal quality is lower, expect some accuracy degradation
- Unique value: combines overnight Muse recording (already supported) with a clinically validated deep learning model
- This is the "killer feature" for health-conscious users — a single number tracking brain health over time

## Complexity Rating: MEDIUM
- Philosopher's Stone is open-source with a Python API — straightforward integration
- Main challenge: validating that Muse 2 overnight recordings produce reliable scores (domain gap from clinical PSG)
- Requires users to wear Muse 2 during sleep (already a supported use case)
- Minimum 4-hour recording needed — partial nights may produce unreliable scores
- pip install philosophers-stone (or clone from GitHub)
- No custom model training needed — uses pre-trained weights
- Estimated implementation: 1 week for integration + validation + UI
